zsh-proxy.plugin.zsh

```diff
@@ -147,17 +167,22 @@ __disable_proxy_git() {
 # NPM
 
 __enable_proxy_npm() {
-    npm config set proxy ${__ZSHPROXY_HTTP}
-    npm config set https-proxy ${__ZSHPROXY_HTTP}
-    yarn config set proxy ${__ZSHPROXY_HTTP} >/dev/null 2>&1
-    yarn config set https-proxy ${__ZSHPROXY_HTTP} >/dev/null 2>&1
+    if which npm >/dev/null; then
+      npm config set proxy ${__ZSHPROXY_HTTP}
+      npm config set https-proxy ${__ZSHPROXY_HTTP}
+      yarn config set proxy ${__ZSHPROXY_HTTP} >/dev/null 2>&1
```

Comment:
What about adding separate detection for `yarn`?

Reply:
Added yarn check as well